### ISSUE TITLE
[core] config::positive() accepts +Inf as a positive value — 103 call sites affected

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#353]
Surfaced during QA of the #184 fix, where a non-finite `alpha_lr` turned out to be reachable through validation.

## The gap

`rlevo-core/src/config.rs`:

```rust
pub fn positive(config: &'static str, field: &'static str, got: f64) -> Result<(), ConfigError> {
    if got > 0.0 { Ok(()) } else { Err(...) }
}
```

`f64::INFINITY > 0.0` is `true`, so **`positive()` accepts `+Inf` as a positive value.** It correctly rejects `NaN` (`NaN > 0.0` is false) and correctly rejects zero and negatives. Infinity is the one hole.

This is not a niche path: `config::positive` has **103 call sites** across the workspace, covering learning rates (`actor_lr`, `critic_lr`, `alpha_lr`, `learning_rate`), physics constants (`cart_mass`, `masspole`, `force_mag`, `link_length_1`, `dt`), and evolution parameters (`initial_sigma`, `d_sigma`, `chi_n`). Every one of them currently treats `+Inf` as valid configuration.

## Why it matters concretely

`SacTrainingConfig::alpha_lr` is validated with `config::positive` (`sac_config.rs:96`). `f64::INFINITY as f32` is `f32::INFINITY` (not NaN), so an infinite `alpha_lr` passes validation and reaches the optimizer.

In `LogAlpha::adam_step` this looked survivable — the `[−88, 88]` clamp appears to catch the resulting `±Inf` parameter. It does not, in general: at `grad == 0` the step is `Inf · 0 = NaN`, and `NaN.clamp(-88, 88)` **propagates** rather than rescuing. Verified by execution:

```
grad=0, lr=inf: step=NaN log_alpha=NaN clamped=NaN
```

#184 handled this defensively at the use site (`adam_step` now rejects a non-finite `lr` alongside a non-finite gradient), but that is one function guarding itself. The other 102 call sites have no such guard, and a physics constant of `+Inf` — `dt`, `masscart`, `force_mag` — will not announce itself as clearly as a learning rate does.

## Reproducer

```rust
// passes validation today
config::positive("SacTrainingConfig", "alpha_lr", f64::INFINITY).unwrap();
```

## Suggested fix

Tighten the predicate to `got > 0.0 && got.is_finite()`, and add the `NotPositive` (or a new `NotFinite`) rejection path for infinity. Worth auditing the sibling validators in the same module at the same time:

- `in_range(lo, hi, got)` is `got >= lo && got <= hi`, which **correctly** rejects `+Inf` for any finite `hi` and correctly rejects `NaN` — but it accepts `+Inf` when called with `hi = f64::INFINITY`. That pattern is already known to be load-bearing: per #173's triage, `qrdqn_config.rs:162` validates `kappa` with `in_range(C, "kappa", 0.0, f64::INFINITY, …)`, and the inclusive lower bound there admits `κ = 0` into a division (filed as #347). The upper bound admits `+Inf` by the same token.
- Whether `ordered(low, high)` should require finiteness of both.

## Scope note

Deliberately filed against `rlevo-core` rather than folded into #184, which is scoped to `sac_alpha.rs`. This is a validation-layer defect with workspace-wide blast radius and belongs on its own terms.

## Related

- #184 — where this surfaced; guards `lr` defensively at the use site
- #347 — `κ = 0` admitted by an inclusive `in_range` lower bound, same module, same class of looseness
- #352 — reward finiteness at the ingestion boundary; the same "non-finite value flows in unchecked" theme at a different layer

**Comment by anthonytorlucci** [comment]
## Correction to this issue's own cross-references

Two "Related" citations in the body above are wrong. Recording rather than silently editing, since both were repeated downstream in `code-review-19-07-2026-core-support.md` §1.1(4):

- **"#347 — `κ = 0` admitted by an inclusive `in_range` lower bound"** → that is **#345**, which is **closed**, fixed by `34d1347`. #347 is an unrelated PPO `log_std` per-head clamp issue.
- **"per #173's triage, `qrdqn_config.rs:162` validates `kappa` with `in_range(C, "kappa", 0.0, f64::INFINITY, …)`"** → #173 is the PPO/PPG Gaussian-head `log_std` clamp issue (also closed), and the `qrdqn` claim is **stale**: since `34d1347`, `qrdqn_config.rs:250` uses `config::positive` plus a hand-rolled `if !(0.5 * kappa * kappa).is_finite()` overflow guard at `:262`.

That hand-rolled guard turned out to be the strongest in-tree argument *for* this fix: a call site had to reimplement finiteness because the shared helper would not carry it.

**Anyone checking the `qrdqn` claim against current source would find the code disagreeing and might reasonably close this issue as stale. It is not stale** — the `config::positive` defect this issue reports was verified present and reproduced by execution.

## Two other numbers

- **"103 call sites"** — 105 today. The drift is configs added since filing, and it is the reason the fix was designed to need no call-site edits at all.
- The code review's **"~10"** `in_range(…, ∞, …)` sites is a significant undercount: the actual figure is **37** across 20 files (40 textual occurrences minus 3 that are `config.rs`'s own rustdoc and definition). That undercount mattered — it made a call-site migration look like a 10-site chore, which is exactly the approach ADR 0060 rejected in favour of tightening the predicate once.
